docs: comprehensive documentation overhaul

SanjayUG opened the pull request:
# Docs: Comprehensive Documentation Overhaul

_Describe the problem or feature in addition to a link to the issues._

This PR overhauls the project's documentation comprehensively by adding structured documentation, detailed guides, extensive code examples, and more.

- Add structured documentation in `docs/` directory
- Create detailed guides for installation, configuration, and usage
- Add extensive code examples and best practices
- Include troubleshooting guides and performance tips
- Improve `README.md` organization and navigation

Closes #19

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added tests (when appropriate).
- [x] You have updated the README or other documentation to account for these changes (when appropriate).
- [x] You have added an entry to the relevant `CHANGELOG.md` (see ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.

[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.0.0/

claymcleod commented:
Hey @SanjayUG,

Unfortunately, this documentation doesn't appear to document how to use Crankshaft—many of the examples won't work, and the configuration options don't match what Crankshaft exposes. If you're interested in working on the issue further, please just ping us here when you've addressed these concerns.

SanjayUG commented:
> Hey @SanjayUG,
> 
> Unfortunately, this documentation doesn't appear to document how to use Crankshaft—many of the examples won't work, and the configuration options don't match what Crankshaft exposes. If you're interested in working on the issue further, please just ping us here when you've addressed these concerns.

No problem, i welcome your feedback.
